rosetta/retriever/retriever_test.go

```diff
@@ -0,0 +1,1196 @@
+			BlockFunc: func(block identifier.Block) (identifier.Block, error) { return block, nil },
+			AccountFunc: func(address identifier.Account) error {
+				return errors.New("invalid account")
+			},
+		}
+		r := &Retriever{
+			validate: validator,
+		}
+
+		_, _, err := r.Balances(testBlockID, testAccount, testCurrencies)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles invalid currency", func(t *testing.T) {
+		validator := &mocks.Validator{
+			BlockFunc:   func(block identifier.Block) (identifier.Block, error) { return block, nil },
+			AccountFunc: func(address identifier.Account) error { return nil },
+			CurrencyFunc: func(currency identifier.Currency) (identifier.Currency, error) {
+				return currency, errors.New("invalid currency")
+			},
+		}
+		r := &Retriever{
+			validate: validator,
+		}
+
+		_, _, err := r.Balances(testBlockID, testAccount, testCurrencies)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles generator failure", func(t *testing.T) {
+		validator := &mocks.Validator{
+			BlockFunc:    func(block identifier.Block) (identifier.Block, error) { return block, nil },
+			AccountFunc:  func(address identifier.Account) error { return nil },
+			CurrencyFunc: func(currency identifier.Currency) (identifier.Currency, error) { return currency, nil },
+		}
+		generator := &mocks.Generator{
+			GetBalanceFunc: func(symbol string) ([]byte, error) {
+				return nil, mocks.DummyError
+			},
+		}
+		r := &Retriever{
+			validate:  validator,
+			generator: generator,
+		}
+
+		_, _, err := r.Balances(testBlockID, testAccount, testCurrencies)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles invoker failure", func(t *testing.T) {
+		validator := &mocks.Validator{
+			BlockFunc:    func(block identifier.Block) (identifier.Block, error) { return block, nil },
+			AccountFunc:  func(address identifier.Account) error { return nil },
+			CurrencyFunc: func(currency identifier.Currency) (identifier.Currency, error) { return currency, nil },
+		}
+		generator := &mocks.Generator{
+			GetBalanceFunc: func(symbol string) ([]byte, error) {
+				return []byte(`testScript`), nil
+			},
+		}
+		invoker := &mocks.Invoker{
+			ScriptFunc: func(height uint64, script []byte, parameters []cadence.Value) (cadence.Value, error) {
+				return nil, mocks.DummyError
+			},
+		}
+		r := &Retriever{
+			validate:  validator,
+			generator: generator,
+			invoke:    invoker,
+		}
+
+		_, _, err := r.Balances(testBlockID, testAccount, testCurrencies)
+		assert.Error(t, err)
+	})
+}
+
+func TestRetriever_Block(t *testing.T) {
+	testHeight := uint64(42)
+	testHeader := &flow.Header{
+		Height:    testHeight,
+		Timestamp: time.Date(1972, 12, 31, 0, 0, 0, 0, time.UTC),
+	}
+	testBlockID := identifier.Block{
+		Index: testHeight,
+		Hash:  "2c4c176c5c095bc3529ab425735077efb2afedd16c9ffc215a898df14fa8ac91",
+	}
+	testTransaction := &object.Transaction{
+		ID: identifier.Transaction{
+			Hash: "0000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
Probably shouldn't use `00000...0000`, because that's the same as `flow.ZeroID`, the zero value for the type.